app/Console/Commands/Sync.php

```diff
@@ -44,12 +45,17 @@ class Sync extends Command
 	public function __construct()
 	{
 		// Fill signature with default values from user configuration
-		$this->signature = sprintf(
-			$this->signature,
-			Configs::getValueAsString('delete_imported', '0'),
-			Configs::getValueAsString('import_via_symlink', '0'),
-			Configs::getValueAsString('skip_duplicates', '0')
-		);
+		try {
+			$this->signature = sprintf(
+				$this->signature,
+				Configs::getValueAsString('delete_imported'),
+				Configs::getValueAsString('import_via_symlink'),
+				Configs::getValueAsString('skip_duplicates')
+			);
+		} catch (ConfigurationKeyMissingException) {
+			// This is necessary other wise we can't even run Composer install
```

Comment:
```suggestion
			// This is necessary, otherwise we can't even run Composer install
```

I'm curious why `composer install` doesn't work without it? Doesn't `composer install` simply download the dependencies based on `composer.lock`? How would the `lychee:sync` console command be affected by any of this?

Or is it that these particular objects are instantiated no matter what when invoking the composer, even if they are not being used?

Reply:
I was curious about that, too. Thank you @kamil4 for checking the changes again. I was too lazy to do it once more. But I should better had to.

Anyway I checked out a fresh clone of the respository, switched to the branch, removed that `catch`-block and looked what would happened. Indeed, we run into a problem here, so @ildyria was right, but the comment is sightly misleading. That's what I meant with precise comments.

The problem is not intrinsic to `composer install` but due to the scripts we automatically invoke after each `composer install`/`composer update`, namely the artisan auto-discovery. The script is invoked via a hook in `postAutoloadDump` and is `php artisan package:discover`.

So at least, the comment needs to be rephrased. Otherwise future developers might get confused. Suggestion:

_Catching exception is necessary as `artisan package:discover` is called after each composer installation/update and artisan tries to instantiate every command_

However, I am not sure if this is the right solution at all. I mean, why does artisan does this anyway. Again some Laravel problem I guess.